### PR TITLE
Fix/multiple concurrent file uploads

### LINKED ISSUE #203: Allow uploading of multiple files outside of chaining `attachFile`
### Current behavior:

Users are able to upload multiple files by chaining `attachFile` methods.

**Example from docs**
```javascript
cy.get('[data-cy="file-input"]')
  .attachFile(yourFixturePath)
  .attachFile(yourBestPicture);
```

### Desired behavior:

The above behavior assumes a static number of known files and removes flexibility from previous versions of passing an array of files. It would be nice to have this feature re-implemented in some way.

**Example with old implementation (trimmed down for brevity)**
```javascript
// uploadMultipleFiles.js

const PDF_FILE = 'Test.pdf';
const DOCX_FILE = 'Test.docx';
const DROPZONE_ELEMENT = '.very-specific-class-name';

Cypress.Commands.add('uploadMultipleFiles', ({ count = 2, type = 'docx' }) => {
  const files = [];

  const fileTypes = { docx: DOCX_FILE, pdf: PDF_FILE };

  for (let index = 0; index < count; index++) {
    if (index % 2 === 0) {
      files.push(fileTypes[type] || fileTypes.docx);
    } else {
      files.push(fileTypes[type] || fileTypes.pdf);
    }
  }

  cy.get(DROPZONE_ELEMENT).attachFile(files, processingOptions);
};

// test.spec.js

describe('A good test', () => {
  it('Should upload 3 files', () => {
    cy.uploadMultipleFiles({ count: 3 });
  };

  it('Should upload 10 DOCX files', () => {
    cy.uploadMultipleFiles({ count: 10, type: 'docx' });
  };
});
```

### Versions

**OS**: macOS Mojave 10.14.6
**Browser**: Chrome 81.0.4044.138
**`cypress`**: `^4.5.0`
**`cypress-file-upload`**: `^4.0.6`

### LINKED ISSUE #234: [Bug] Regression: Multiple files upload is no more available since 4.0.0
### Current behavior:

Single file upload with attachFile(). In previous versions upload / uploadFiles was available for multiple file upload

### Desired behavior:

provide attachFiles() in order to be able to migrate code

### PR DESCRIPTION
**Pull request by paulblyth**
Hey @abramenal - hoping this is an acceptable changeset. I know there's an open PR (#227) but that looks like it has gone stale and looks like it might have had some issues.

Closes #203 
Closes #234 

There's a little bit of ES6 in here and I've not contributed to a Cypress plugin before so let me know if that's bad practice - and please be extra critical :)

Note: I also had to commit changes in the recipes using `--no-verify` due to some `eslint` errors, I didn't want to dig too much into those

### CODE REVIEW

**Comment by paulblyth** [comment]
@tommy-anderson [I saw your comment](https://github.com/abramenal/cypress-file-upload/pull/227#issuecomment-760141747) - would you be able to give these changes a test drive in your migration?

File: README.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ import 'cypress-file-upload';
 ```
 
 Note: With Typescript, ensure the following is in your `cypress\tsconfig.json` file:
+
```
Comment (paulblyth) [inline README.md:38]: My linter must have gotten a little eager, happy to revert the formatting changes if needed

**Comment by abramenal** [comment]
I've made small changes and tested it all locally. Great job @paulblyth, thank you!
Next steps:
- merge to master
- wait for CI to be green (it will check the logic against some browsers in headed and headless modes)
- release it tomorrow, boom! 💥

**Comment by abramenal** [comment]
@all-contributors add @paulblyth for code

**Comment by abramenal** [comment]
Released with `v4.2.0`

**Comment by paulblyth** [comment]
@abramenal thanks for cleaning up and merging 🙌

**Comment by abramenal** [comment]
Released in v5 btw

**Pull request merged**